data_write.py

```diff
@@ -39,83 +98,43 @@
-        csv_read = csv.reader(f1, delimiter=',')
-        for i, row in enumerate(csv_read):
-            if i == 1:
-                seed = str(row[0])
-            elif i > 4:
-                individual.append(row[0])
-                parent1.append(row[1])
-                parent2.append(row[2])
-                opperator.append(row[3])
-    f1.close()
+# read in parents
+read_parents(g.generation, g.population, individual,
+             parent1, parent2, opperator, seed)
 
 # Read in Chi score
-with open('chiScores.csv') as f2:
-    csv_read = csv.reader(f2, delimiter=',')
-    for i, row in enumerate(csv_read):
-        if i > 1:
-            chi.append(str(round(float(row[0]), 8)))
-f2.close()
+filename = "chiScores.csv"
+read_chi(chi, filename)
 
 # Read in fitness score and error
-with open('fitnessScores.csv') as f3:
-    csv_read = csv.reader(f3, delimiter=',')
-    for i, row in enumerate(csv_read):
-        if i > 1:
-            fitness.append(str(round(float(row[0]), 8)))
-            error.append(str(round(float(row[1]), 8)))
-f3.close()
+filename = "fitnessScores.csv"
+read_fitness(fitness, error, filename)
 
 # Read in DNA
-with open('generationDNA.csv') as f4:
-    csv_read = csv.reader(f4, delimiter=',')
-    individual_no = 0
-    for i, row in enumerate(csv_read):
-        if i > 8:
-            if (i-9) % sections == 0:
-                j = 0
-            if (i-9) % sections != 0:
-                j = (i-9) % sections
-            for k in range(genes):
-                dna[individual_no][j][k] = float(row[k])
-            if j == (i-9) % sections and (i-9) % sections != 0:
-                individual_no = individual_no + 1
-            if sections == 1:
-                individual_no = individual_no + 1
-f4.close()
+filename = "generationDNA.csv"
+read_dna(sections, genes, dna, filename)
 
 # Write Data into file
 with open('generationData.csv', "w") as f5:
     # Write out the header
-    f5.write("Generation Data for generation " + str(g.generation) + '\n')
-    f5.write(seed + '\n')
+    f5.write(f"Generation Data for generation {g.generation}\n")
+    f5.write(f"{seed}\n")
     f5.write('\n')
     f5.write("Individual, Chi, Fitness, Error, Parent 1, Parent 2, Opperator")
     for x in range(sections*genes):
-        f5.write(", Gene " + str(x+1))
+        f5.write(f", Gene {x+1}")
     f5.write('\n')
 
     # Write data
     for i in range(len(individual)):
-        f5.write(str(individual[i]))
-        f5.write(", " + str(chi[i]))
-        f5.write(", " + str(fitness[i]))
-        f5.write(", " + str(error[i]))
-        f5.write(", " + str(parent1[i]))
-        f5.write(", " + str(parent2[i]))
-        f5.write(", " + str(opperator[i]))
+        f5.write(f"{individual[i]}")
+        f5.write(f", {chi[i]}")
+        f5.write(f", {fitness[i]}")
+        f5.write(f", {error[i]}")
+        f5.write(f", {parent1[i]}")
+        f5.write(f", {parent2[i]}")
+        f5.write(f", {opperator[i]}")
```

Comment:
```suggestion
        f5.write(f"{individual[i]}, {chi[i]}, {fitness[i]}, {error[i]}, "
                 f"{parent1[i]}, {parent2[i]}, {opperator[i]}")
```
This is shorter, more readable, and only does one `write` operation.